### ISSUE TITLE
Min/Max ctrl function issue

### ISSUE THREAD

**Issue by jutrasi** [firemodels/fds#7588]
Hi,

I am trying to use the min/max control functions with a set of devices however for both functions they only seem to output the first input value regardless of its value. I have included a simple example below just using in MAX function but the same holds true to the MIN function. I am not sure if I am missing something or not. Thank you for your help.

`&HEAD CHID='Max_test',TITLE='Max_test'/

&MESH IJK=30,30,20, XB=-3,3,-3,3,0,4/

&TIME T_END=5/

&DEVC ID='Timer',XYZ=0,0,1, QUANTITY='TIME'/

&CTRL ID='T1', FUNCTION_TYPE='MULTIPLY', INPUT_ID='Timer','CONSTANT',CONSTANT=1/
&DEVC ID='T1v', QUANTITY='CONTROL VALUE', CTRL_ID='T1'/
&CTRL ID='T2', FUNCTION_TYPE='MULTIPLY', INPUT_ID='Timer','CONSTANT',CONSTANT=4/
&DEVC ID='T2v', QUANTITY='CONTROL VALUE', CTRL_ID='T2'/
&CTRL ID='T3', FUNCTION_TYPE='MULTIPLY', INPUT_ID='Timer','CONSTANT',CONSTANT=2/
&DEVC ID='T3v', QUANTITY='CONTROL VALUE', CTRL_ID='T3'/

&CTRL ID='Max1', FUNCTION_TYPE='MAX', INPUT_ID='T3','T2','T1'/
&DEVC ID='Max1v', QUANTITY='CONTROL VALUE', CTRL_ID='Max1'/
&CTRL ID='Max1-1', FUNCTION_TYPE='MAX', INPUT_ID='T1','T2','T3'/
&DEVC ID='Max1-1v', QUANTITY='CONTROL VALUE', CTRL_ID='Max1-1'/

&VENT MB='XMIN', SURF_ID='OPEN' /
&VENT MB='XMAX', SURF_ID='OPEN' /
&VENT MB='YMIN', SURF_ID='OPEN' /
&VENT MB='YMAX', SURF_ID='OPEN' /

&TAIL/`

**Comment by rmcdermo** [comment]
Jason,
This may be better suited for the discussion forum, but please have a quick look.  Thanks

**Comment by rmcdermo** [comment]
Jason,
I reproduced the issue.  It looks like a bug to me.  It seems you always point to `DV => DEVICE(CF%INPUT(1))` for CF_MAX case in ctrl.f90

**Comment by rmcdermo** [comment]
Jason,
If I do this, then it works.  Want me to push this up?
```
CASE (CF_MAX)
      CF%INSTANT_VALUE=-HUGE(TWO_EPSILON_EB)

      DO NC = 1, CF%N_INPUTS
         SELECT CASE (CF%INPUT_TYPE(1))
            CASE (DEVICE_INPUT)
               DV => DEVICE(CF%INPUT(NC))
               CF%INSTANT_VALUE = MAX(DV%SMOOTHED_VALUE,CF%INSTANT_VALUE)
               CF%MESH = DV%MESH
            CASE (CONTROL_INPUT)
               IF (.NOT. CONTROL(CF%INPUT(NC))%UPDATED) THEN
                  CALL EVALUATE_CONTROL(T,CF%INPUT(NC),DT,CTRL_STOP_STATUS)
                  CF => CONTROL(ID)
               ENDIF
               CF%INSTANT_VALUE = MAX(CF%INSTANT_VALUE,CONTROL(CF%INPUT(NC))%INSTANT_VALUE)
               CF%MESH=CONTROL(CF%INPUT(NC))%MESH
            CASE (CONSTANT_INPUT)
               CF%INSTANT_VALUE = MAX(CF%INSTANT_VALUE,CF%CONSTANT)
         END SELECT
      ENDDO
```

**Comment by drjfloyd** [comment]
That looks right.  Same structure for CF_MIN would be needed. Thanks!

**Comment by rmcdermo** [comment]
Will do.

**Comment by rmcdermo** [comment]
Should be fixed with PR #7606 

@jutrasi Thanks for the nice bug report.  I'm nominating this for a Boris Stock award!  This will be fixed on next release, out in the next couple weeks.

**Comment by jutrasi** [comment]
Thanks for looking into this/ fixing it!
